### PR TITLE
Include component resources to be installed with chart

### PR DESCRIPTION
**Pull request by mjakobczyk**
**Description**:
- deploy `istio-configuration` k8s resources after performing Istio installation
- omit `IstioOperator` resource from manifest to avoid duplication

### CODE REVIEW

File: pkg/reconciler/instances/istio/action.go [hunk 1]
```diff
@@ -41,5 +42,31 @@ func (a *ReconcileAction) Run(version, profile string, config map[string]interfa
...
+	_, err = context.KubeClient.Deploy(context.Context, generateNewManifestWithoutIstioOperatorFrom(manifest.Manifest), istioNamespace, nil)
+	if err != nil {
+		return errors.Wrap(err, "Could not deploy Istio resources")
+	}
+
 	return nil
 }
+
+func generateNewManifestWithoutIstioOperatorFrom(manifest string) string {
+	separator := "---"
```
Comment (pbochynski) [inline pkg/reconciler/instances/istio/action.go:54]: It smells. Can't you parse it (or use k8s client to do it) and remove IstioOperator resource from the list?

**Review by pbochynski — CHANGES_REQUESTED** [review]
String manipulation is not the best idea for managing YAML files.

File: pkg/reconciler/instances/istio/action.go (hunk 1 above)
Comment (tobiscr) [reply pkg/reconciler/instances/istio/action.go:54]: You can use our K8s-API features to convert manifests into `[]unstructured.Unstructured` entities and operate on them... see https://github.com/kyma-incubator/reconciler/blob/main/pkg/reconciler/kubernetes/kubeclient/unstructured.go#L15

**Review by tobiscr — CHANGES_REQUESTED** [review]

File: pkg/reconciler/instances/istio/action.go [hunk 2]
```diff
@@ -41,5 +44,36 @@ func (a *ReconcileAction) Run(version, profile string, config map[string]interfa
 		return errors.Wrap(err, "Could not patch MutatingWebhookConfiguration")
 	}
 
+	generated := generateNewManifestWithoutIstioOperatorFrom(manifest.Manifest)
+	fmt.Println(generated)
```
Comment (tobiscr) [inline pkg/reconciler/instances/istio/action.go:48]: Probably a left over? If not, please use the  provided `logger` instance from the `ActionContext`.

File: pkg/reconciler/instances/istio/action.go [hunk 3]
```diff
@@ -41,5 +44,36 @@ func (a *ReconcileAction) Run(version, profile string, config map[string]interfa
...
+		if unstruct.GetKind() == "IstioOperator" {
+			continue
+		}
+
+		unstructBytes, err := unstruct.MarshalJSON()
+		if err != nil {
+			return ""
+		}
+
+		builder.WriteString("---\n")
```
Comment (tobiscr) [inline pkg/reconciler/instances/istio/action.go:74]: Would recommend not to serialize each `Unstructured` one by one and add YAML delimiters (`---`) manually to it. Better to rely completely on the the K8s API to convert it back to YAML.

Therefore, an option could be to drop just the element from the slice of `Unstructured` entities. Like

      var withoutOperator []unstructured.Unstructured
      for _, unstr := range unstructs {
         if unstruct.GetKind() == "IstioOperator" {
            continue
         }
         withoutOperator = append(withoutOperator, unstr)
      }
      
Finally, serialise it, e.g. by using `unstructured.UnstructuredJSONScheme.Encode()` ([link to API](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go#L347-L351))

**Review by tobiscr — CHANGES_REQUESTED** [review]

File: pkg/reconciler/instances/istio/action.go (hunk 3 above)
Comment (tobiscr) [reply pkg/reconciler/instances/istio/action.go:74]: It's actually the opposite of this call: https://github.com/kyma-incubator/reconciler/blob/main/pkg/reconciler/kubernetes/kubeclient/unstructured.go#L91

Comment (tobiscr) [reply pkg/reconciler/instances/istio/action.go:74]: If you like, feel free to add this new function for converting `unstructured.Unstructured` to our API https://github.com/kyma-incubator/reconciler/blob/main/pkg/reconciler/kubernetes/kubeclient/unstructured.go ;)

**Review by pbochynski — APPROVED** [review]

**Review by tobiscr — DISMISSED** [review]

**Review by tobiscr — APPROVED** [review]

**Pull request merged**